sim/aviron/src/lib/memory.zig

```diff
@@ -0,0 +1,194 @@
+const std = @import("std");
+const io = @import("io.zig");
+
+/// Use the unified device interface
+pub const Backend = io.Device;
+const Device = io.Device;
+
+/// A mapping entry within a MemorySpace.
+pub const Segment = struct {
+    /// Base address within the enclosing memory space.
+    at: usize,
+    /// Size in bytes of the mapped range.
+    size: usize,
```

Comment:
As above: `usize` is for host memory sizes, not for AVR memory sizes. Imagine someone using a 16 or 20 bit host, but aviron would expect to fit 24 bits of memory here.